post_office/mail.py

```diff
@@ -270,7 +272,35 @@ def send(email):
     Email.objects.filter(id__in=email_ids).update(status=STATUS.sent)
 
     email_ids = [email.id for (email, e) in failed_emails]
-    Email.objects.filter(id__in=email_ids).update(status=STATUS.failed)
+
+    requeued_emails = 0
+    # Retry is enabled
+    if get_max_retries():
+        max_retries = get_max_retries()
+        emails_failed = Email.objects.filter(id__in=email_ids)
+
+        emails_max_retries_exceeded = []  # emails status will change to failed
+        emails_to_requeue = []  # emails status will change to requeued
+
+        for email in emails_failed:
+            if email.number_of_retries < max_retries:
+                email.status = STATUS.requeued
+                email.scheduled_time = now() + get_time_delta_to_retry()
+                if email.number_of_retries is None:
```

Comment:
We shouldn't need to check for `None` since the `number_of_retries` column is not nullable.

Reply:
So we either change this, or make the `number_of_retries` column nullable, which may speed up schema migrations on dataabases and also consume less space (I think).

Reply:
I put in the null column and remove defaut, but I had to refactor the code a little bit.